x-pack/legacy/plugins/canvas/public/plugin.ts

```diff
@@ -0,0 +1,91 @@
+ */
+
+import React from 'react';
+import ReactDOM from 'react-dom';
+import { Chrome } from 'ui/chrome';
+import { IModule } from 'angular';
+import { Storage } from 'ui/storage';
+import { CoreSetup, CoreStart, Plugin } from '../../../../../src/core/public';
+// @ts-ignore: Untyped Local
+import { initStateManagement, initLocationProvider } from './angular/config';
+import { CanvasRootControllerFactory } from './angular/controllers';
+// @ts-ignore: Untypled Local
+import { initStore } from './angular/services';
+// @ts-ignore: untyped local component
+import { HelpMenu } from './components/help_menu/help_menu';
+// @ts-ignore: untyped local
+import { loadExpressionTypes } from './lib/load_expression_types';
+// @ts-ignore: untyped local
+import { loadTransitions } from './lib/load_transitions';
+
+// @ts-ignore: untyped local
+import { initClipboard } from './lib/clipboard';
+
+export { CoreStart };
+/**
+ * These are the private interfaces for the services your plugin depends on.
+ * @internal
+ */
+// This interface will be built out as we require other plugins for setup
+export interface CanvasSetupDeps {} // eslint-disable-line @typescript-eslint/no-empty-interface
+export interface CanvasStartDeps {
+  __LEGACY: {
+    absoluteToParsedUrl: (url: string, basePath: string) => any;
+    formatMsg: any;
+    loadingCount: any;
+    badge: Chrome['badge'];
+    setRootController: Chrome['setRootController'];
+    storage: typeof Storage;
+    trackSubUrlForApp: Chrome['trackSubUrlForApp'];
+    uiModules: {
+      get: (module: string) => IModule;
+    };
+  };
+}
+
+/**
+ * These are the interfaces with your public contracts. You should export these
+ * for other plugins to use in _their_ `SetupDeps`/`StartDeps` interfaces.
+ * @public
+ */
+// These interfaces are empty for now but will be populate as we need to export
+// things for other plugins to use at startup or runtime
+export interface CanvasSetup {} // eslint-disable-line @typescript-eslint/no-empty-interface
+export interface CanvasStart {} // eslint-disable-line @typescript-eslint/no-empty-interface
+
+/** @internal */
+export class CanvasPlugin
+  implements Plugin<CanvasSetup, CanvasStart, CanvasSetupDeps, CanvasStartDeps> {
+  public setup(core: CoreSetup, plugins: CanvasSetupDeps) {
+    // This is where any setup actions need to occur.
+    // Things like registering functions to the interpreter that need
+    // to be available everywhere, not just in Canvas
+    return {};
+  }
+
+  public start(core: CoreStart, plugins: CanvasStartDeps) {
+    loadExpressionTypes();
+    loadTransitions();
+
+    initStateManagement(core, plugins);
+    initLocationProvider(core, plugins);
+    initStore(core, plugins);
+    initClipboard(plugins.__LEGACY.storage);
+
+    const CanvasRootController = CanvasRootControllerFactory(core, plugins);
+    plugins.__LEGACY.setRootController('canvas', CanvasRootController);
```

Comment:
I think this can be replaced by the compatibility layer we have in place for the application service which does essentially the exact same thing. The API you would use is `core.application.register` from your setup method.